lib/wrap-request.js

```diff
@@ -0,0 +1,59 @@
+/* eslint padded-blocks: 0 */
+module.exports = wrapRequest
+
+const noop = () => Promise.resolve()
+
+function wrapRequest (state, emitter, request, options, retries = 0) {
+  const retryRequest = function (retryAfter) {
+    return new Promise(resolve => setTimeout(resolve, retryAfter * 1000))
+      .then(() => wrapRequest(state, emitter, request, options, retries + 1))
+  }
+
+  const jobOptions = retries > 0 ? { weight: 0, priority: 0 } : {}
+  return state.globalLimiter.schedule(jobOptions, async function () {
+
+    // Guarantee at least 1000ms between writes
+    if (options.method !== 'GET' && options.method !== 'HEAD') {
+      await state.writeLimiter.schedule(noop)
+    }
+
+    // Guarantee at least 5000ms between requests that trigger notifications
+    if (state.triggersNotification[options.url]) {
+      await state.triggersNotificationLimiter.schedule(noop)
+    }
```

Comment:
I think we should run either `writeLimiter.schedule` or `triggersNotificationLimiter.schedule`, but not both? We also shouldn’t run `triggersNotificationLimiter.schedule` for `GET`/`HEAD` requests, only for mutating requests

Reply:
It's easier to just do both and let Bottleneck figure it out. If you already waited 1000ms in the first limiter, the second will calculate that it only needs to make you wait 4000ms. It's cleaner code that way. We write the **guarantees** we want and let the library do the work.